test/libsolidity/syntaxTests/viewPureChecker/inline_assembly_instructions_disallowed_paris.sol

```diff
@@ -0,0 +1,12 @@
+contract C {
+    function f() public {
+        assembly {
+            pop(difficulty())
+        }
+    }
+}
+// ====
+// EVMVersion: >=paris
+// ----
+// DeclarationError 4619: (74-84): Function "difficulty" not found.
```

Comment:
I'd put them in `syntaxTests/inlineAssembly/`.

Looks like you put them in `viewPureChecker` because it already has other tests checking for allowed/disallowed instructions but those tests specifically check this in context of a function being pure/view/mutable (which should be caught by `ViewPureChecker`). Your tests have nothing to do with that. They're general inline assembly tests.